ios-target-app/Home/Views/SaveTargetBottomSheetViewController.swift

```diff
@@ -41,16 +44,54 @@ class SaveTargetBottomSheetViewController: UIViewController {
   
   private lazy var saveButton = UIButton(
     text: "home_save_button_label".localized,
-    target: self
+    target: self,
+    action: #selector(saveTargetButtonTapped)
   )
   
+  private var cancellables: Set<AnyCancellable> = []
+  
+  init(viewModel: TargetViewModel) {
+    self.viewModel = viewModel
+    super.init(nibName: nil, bundle: nil)
+    
+    setupSubscribers()
+  }
+  
+  @available(*, unavailable)
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
   override func viewDidLoad() {
     super.viewDidLoad()
     
     setBottomSheetUIConfigs()
     configureViews()
   }
   
+  private func setupSubscribers() {
+    viewModel.statePublisher
+      .sink { [weak self] state in
+        self?.setViewsState(state: state)
+      }.store(in: &cancellables)
+  }
+  
+  private func setViewsState(state: TargetViewModelState) {
+    switch state {
+    case .created:
+      self.dismiss(animated: true, completion: nil)
```

Comment:
Avoid the use of self when not explicitly needed.
```suggestion
      dismiss(animated: true, completion: nil)
```